src/interfaces/IRequiredConfigOptions.ts

```diff
@@ -35,5 +35,6 @@ export interface IRequiredConfigOptions extends IConfigOptions {
 	schemaFileName: string;
 	ajvOptions: Ajv.Options;
 	useDotNotation: boolean;
+	dotNotationSeparator: string;
```

Comment:
I am confusing this options is importing IConfigOptions. First you set it as optional type that can use and then it is defined as needed operator. Do you use IConfigOption as extern options and IRequiredConfigOptions as intern requirement?